soc/silabs/silabs_siwx91x/siwg917/siwx91x_nwp.c

```diff
@@ -318,17 +334,29 @@ static int siwx91x_get_nwp_config(const struct device *dev,
 					   SL_SI91X_FEAT_SECURITY_PSK | SL_SI91X_FEAT_AGGREGATION |
 					   SL_SI91X_FEAT_HIDE_PSK_CREDENTIALS,
 			.tcp_ip_feature_bit_map = SL_SI91X_TCP_IP_FEAT_EXTENSION_VALID,
-			.custom_feature_bit_map = SL_SI91X_CUSTOM_FEAT_EXTENSION_VALID |
-						  SL_SI91X_CUSTOM_FEAT_ASYNC_CONNECTION_STATUS |
-						  SL_SI91X_CUSTOM_FEAT_RTC_FROM_HOST,
+			.custom_feature_bit_map =
+				IS_ENABLED(CONFIG_WIFI_SILABS_SIWX91X)
+					? (SL_SI91X_CUSTOM_FEAT_SOC_CLK_CONFIG_160MHZ |
+					   SL_SI91X_CUSTOM_FEAT_EXTENSION_VALID)
+					: (SL_SI91X_CUSTOM_FEAT_EXTENSION_VALID |
+					   SL_SI91X_CUSTOM_FEAT_ASYNC_CONNECTION_STATUS |
+					   SL_SI91X_CUSTOM_FEAT_RTC_FROM_HOST),
 			.ext_custom_feature_bit_map =
-				SL_SI91X_EXT_FEAT_XTAL_CLK | SL_SI91X_EXT_FEAT_1P8V_SUPPORT |
-				SL_SI91X_EXT_FEAT_DISABLE_XTAL_CORRECTION |
-				SL_SI91X_EXT_FEAT_NWP_QSPI_80MHZ_CLK_ENABLE |
-				SL_SI91X_EXT_FEAT_FRONT_END_SWITCH_PINS_ULP_GPIO_4_5_0 |
-				SL_SI91X_EXT_FEAT_FRONT_END_INTERNAL_SWITCH |
-				SL_SI91X_EXT_FEAT_XTAL_CLK,
-		}
+				IS_ENABLED(CONFIG_WIFI_SILABS_SIWX91X)
+					? SL_SI91X_EXT_FEAT_FRONT_END_SWITCH_PINS_ULP_GPIO_4_5_0
+					: (SL_SI91X_EXT_FEAT_XTAL_CLK |
+					   SL_SI91X_EXT_FEAT_1P8V_SUPPORT |
+					   SL_SI91X_EXT_FEAT_DISABLE_XTAL_CORRECTION |
+					   SL_SI91X_EXT_FEAT_NWP_QSPI_80MHZ_CLK_ENABLE |
+					   SL_SI91X_EXT_FEAT_FRONT_END_SWITCH_PINS_ULP_GPIO_4_5_0 |
+					   SL_SI91X_EXT_FEAT_FRONT_END_INTERNAL_SWITCH |
+					   SL_SI91X_EXT_FEAT_XTAL_CLK),
+		},
+	#if IS_ENABLED(CONFIG_WIFI_SILABS_SIWX91X)
+		.ta_pool = {.tx_ratio_in_buffer_pool = TX_POOL_RATIO,
+					.rx_ratio_in_buffer_pool = RX_POOL_RATIO,
+					.global_ratio_in_buffer_pool = GLOBAL_POOL_RATIO}
+	#endif
```

Comment:
This struct is already a `default_config`, you don't need to define any extra symbols for the value inside (the default value for the default value). So, you can drop the `*_POOL_RATIO` definitions.

Reply:
BTW, do we want to set these values only with Wifi? If yes, avoid `#if`. Prefer either `IF_ENABLED(CONFIG_WIFI_SILABS_SIWX91X, 1, 0)` or `if (IS_ENABLED(CONFIG_WIFI_SILABS_SIWX91X)) { ... }`.